jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -88,18 +88,43 @@ public LayerEntry(
    * @param permissions the file permissions on the container. Use {@code null} to use defaults (644
    *     for files, 755 for directories)
    */
+  public LayerEntry(Path sourceFile, AbsoluteUnixPath extractionPath, FilePermissions permissions) {
+    this(
+        sourceFile,
+        extractionPath,
+        permissions,
+        DEFAULT_FILE_TIMESTAMP_PROVIDER.apply(sourceFile, extractionPath));
+  }
+
+  /**
+   * Instantiates with a source file and the path to place the source file in the container file
+   * system.
+   *
+   * @param sourceFile the source file to add to the layer
+   * @param extractionPath the path in the container file system corresponding to the {@code
+   *     sourceFile}
+   * @param permissions the file permissions on the container
+   * @param modifiedTime the file modification time, default to 1 second since the epoch
+   *     (https://github.com/GoogleContainerTools/jib/issues/1079)
+   */
   public LayerEntry(
-      Path sourceFile, AbsoluteUnixPath extractionPath, @Nullable FilePermissions permissions) {
-    this(sourceFile, extractionPath, permissions, 1000);
+      Path sourceFile,
+      AbsoluteUnixPath extractionPath,
+      FilePermissions permissions,
+      Instant modifiedTime) {
+    this.sourceFile = sourceFile;
+    this.extractionPath = extractionPath;
+    this.permissions = permissions;
+    this.modifiedTime = modifiedTime;
   }
 
   /**
    * Returns the modification time of the file in the entry.
    *
    * @return the modification time
    */
-  public long getLastModifiedTime() {
-    return lastModifiedTime;
+  public Instant getLastModifiedTime() {
+    return modifiedTime;
```

Comment:
nit: `modifiedTime`, `lastModified`, and `lastModifiedTime` are used throughout, so let's try to be consistent. I think "modification time" in Javadocs is OK though.